crates/nu_plugin_polars/src/dataframe/command/core/open.rs

```diff
@@ -165,6 +166,17 @@ fn command(
 
     let hive_options = build_hive_options(plugin, call)?;
 
+    let uri = spanned_file.item.clone();
+    let data_source = if resource.cloud_options.is_some() {
+        Url::from_str(&uri)
+            .map(|url| DataSource::Url(url.into()))
+            .unwrap_or_else(|_| DataSource::FilePath(uri.into()))
```

Comment:
I am not as familiar as @ayax79 if there may be other paths or if `cloud_options` is the clear indicator on whether it is a local or uri path.

Reply:
Currently, cloud options is created if it is a supported cloud uri... right now only s3://. If it isn't an s3 resource, then cloud_options will be None. This should be fine. Really, we could support Url across the board at some point and provide a file:// url.. I wouldn't worry about it for now.

I'll be adding to other cloud providers at some point.